Batch File/Batch File.sublime-syntax

```diff
@@ -161,7 +161,7 @@ contexts:
   comment-body:
     - meta_scope: comment.line.colon.dosbatch
     - include: line-continuations
-    - match: \n
+    - match: $
```

Comment:
Generally the trailing new line char in line comments is deliberately scoped as a comment so that auto-complete  and snippets etc wouldn't trigger.

What problem are you actually solving here? Do.many of the new test assertions fail without this change?

Did you consider:
```suggestion
    - match: $\n?
```

Reply:
`\n` or `$\n?` shouldn't make any different practically.

That said, `\n` being part of the `comment` scope is indeed on purpose.

Reply:
> What problem are you actually solving here? Do.many of the new test assertions fail without this change?

There are two issues:
 - multiple colons followed immediately by a non-empty line cause said line to be scoped as a comment
 - `\n` on an empty line can itself be escaped, making the following lines incorrectly scoped

> Did you consider:

Thanks for the tip, it slipped my mind. That solves the first issue.

Reply:
> multiple colons followed immediately by a non-empty line cause said line to be scoped as a comment

... by intention as the linked ss64 site defines it as such.

see: https://github.com/sublimehq/Packages/blob/3b7c0d35953d66d452fa139fdcc0a3f3689dae15/Batch%20File/Batch%20File.sublime-syntax#L156

Reply:
With regards to `\n` vs. `$\n?`, the former one is enough when using batch scripts on its own, the latter one may be more safe/reliable in cases of batch being extended in templating syntaxes, which may inject interpolation contexts into comments, which may already consume `\n`.

Reply:
> > multiple colons followed immediately by a non-empty line cause said line to be scoped as a comment
> 
> ... by intention as the linked ss64 site defines it as such.
> 
> see:
> 
> https://github.com/sublimehq/Packages/blob/3b7c0d35953d66d452fa139fdcc0a3f3689dae15/Batch%20File/Batch%20File.sublime-syntax#L156

I've read that multiple times and don't know what you're alluding to. It's plain wrong anyway:

```batchfile
@echo off
::: comment.bat
:::
echo Not a comment
```

```console
> comment.bat
Not a comment
```

```batchfile
@echo off
rem comment2.bat
rem
echo Not a comment
```

```console
> comment2.bat
Not a comment
```

Reply:
Snippet 1 is not handled correctly, currently. `echo` must not be scoped comment. `:: ` is ignored as not being a valid label or command. These may not be comments in the first place, but seem to be one of those common constructs for such. I actually only use these kinds rather than horrible "rem".

Snippet 2 and 4 show valid redirections, which may not have an effect due to nothing being in front of them. Those are however no comments.

Snippet 4 is working as expected.

Reply:
"Snippets" 2 and 4 are just console invocations of snippets 1 and 3.

I'm using them in favor of `rem` as well, though not everywhere - in some places `:::` is invalid (such as in parenthesized blocks).

Note that `:::` is slightly safer than `::` due to `subst`.